### PR TITLE
rewrite code asset with asset blocks

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* rewriting the code asset with asset blocks
* this pr is mostly a rewrite of the tera-code-dynamic component to use the AssetBlock component

### CODE REVIEW

File: packages/client/hmi-client/src/components/drilldown/tera-drilldown.vue [hunk 1]
```diff
@@ -90,7 +90,7 @@ than the main application behind the modal when these render issues come, howeve
 
 main {
 	flex-grow: 1;
-	padding: 0 0 0 var(--gap-small);
+	padding: var(--gap-small);
```
Comment (blanchco) [inline packages/client/hmi-client/src/components/drilldown/tera-drilldown.vue:93]: add padding back to drilldown

File: packages/client/hmi-client/src/components/widgets/tera-asset-block.vue [hunk 2]
```diff
@@ -28,11 +34,25 @@ import InputSwitch from 'primevue/inputswitch';
...
+		type: Boolean,
+		default: true
+	},
+	isIncluded: {
+		type: Boolean
+	},
+	isEditable: {
+		type: Boolean
+	},
+	toggleable: {
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/widgets/tera-asset-block.vue:51]: `isToggleable` > `toggleable`?

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/components/widgets/tera-asset-block.vue [hunk 3]
```diff
@@ -28,11 +34,25 @@ import InputSwitch from 'primevue/inputswitch';
...
-defineProps<{
-	isDeletable?: boolean;
-	isIncluded?: boolean;
-	isEditable?: boolean;
-}>();
+defineProps({
+	isDeletable: {
+		type: Boolean
+	},
+	isIncludable: {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/widgets/tera-asset-block.vue:41]: What about `isPermitted` or `isSuitable`?

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/widgets/tera-asset-block.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/components/widgets/tera-asset-block.vue]: changed!

**Pull request merged**